p2pool/src/sharechain/in_memory.rs

```diff
@@ -666,8 +670,30 @@ impl ShareChain for InMemoryShareChain {
             PowAlgorithm::Sha3x => Difficulty::from_u64(self.minimum_sha3_target_difficulty).unwrap(),
         };
 
-        let difficulty = chain_read_lock.lwma.get_difficulty().unwrap_or(Difficulty::min());
-        let difficulty = cmp::max(min, difficulty);
+        let difficulty = match chain_read_lock.lwma.get_difficulty() {
+            Some(val) => {
+                if val < min {
+                    debug!(
+                        target: LOG_TARGET,
+                        "[{:?}] Calculated difficulty ({}) too low, using the minimum ({}), likely due to insufficient \
+                        timestamps({})",
```

Comment:
```suggestion
                        "[{:?}] Calculated difficulty ({}) too low, using the minimum ({})",
```
correlation does not imply causation, this part of the message will just cause confusion, I think leave it out